src/main/java/edu/harvard/dbmi/avillach/dictionaryetl/columnmeta/AbstractColumnMetaProcessor.java

```diff
@@ -0,0 +1,297 @@
+    protected volatile boolean running = true;
+
+    public AbstractColumnMetaProcessor(ColumnMetaMapper columnMetaMapper) throws SQLException {
+        this.columnMetaMapper = columnMetaMapper;
+    }
+
+    /**
+     * Initializes and starts the processing threads needed for handling column metadata.
+     * This method is called by {@link #processColumnMetaCSV(String, String)} before processing begins.
+     * 
+     * <p>Implementations should:</p>
+     * <ul>
+     *   <li>Create and start any necessary threads for processing</li>
+     *   <li>Set up any required consumers for the {@link #processedColumnMetas} queue</li>
+     *   <li>Ensure threads are properly configured (e.g., as daemon threads if appropriate)</li>
+     * </ul>
+     * 
+     * <p>The implementation should continue processing until {@link #running} is set to false.</p>
+     * 
+     * @see #processColumnMetaCSV(String, String)
+     * @see #processColumnMetas(List)
+     */
+    protected abstract void startProcessing();
+
+    /**
+     * Creates and returns an {@link ExecutorService} to be used for parallel processing of column metadata.
+     * This method is called during the initialization of the {@link AbstractColumnMetaProcessor}.
+     *
+     * <p>Implementations should:</p>
+     * <ul>
+     *   <li>Create an appropriate thread pool based on system resources or application requirements</li>
+     *   <li>Consider database connection limits if database operations will be performed</li>
+     *   <li>Return a non-null {@link ExecutorService} instance</li>
+     * </ul>
+     *
+     * <p>The returned {@link ExecutorService} will be used for processing column metadata in parallel.</p>
+     *
+     * @return A non-null {@link ExecutorService} instance to be used for parallel processing
+     * @throws IllegalStateException if the implementation returns null
+     */
+    protected abstract ExecutorService createThreadPool();
+
+    /**
+     * Uses the columnMeta.csv that is created as part of the HPDS ETL to hydrate the data-dictionary database.
+     * The CSV file is expected to exist at /opt/local/hpds/columnMeta.csv.
+     */
+    public String processColumnMetaCSV(String csvPath, String errorFile) throws RuntimeException {
+        this.fixedThreadPool = createThreadPool();
+        if (this.fixedThreadPool == null) {
+            throw new IllegalStateException("Subclass must provide a non-null ExecutorService");
+        }
+
+        if (errorFile == null) {
+            errorFile = "/opt/local/hpds/columnMetaErrors.csv";
+        } else if (!errorFile.endsWith(".csv")) {
+            return "The error file must be a csv.";
+        }
+
+        if (csvPath == null) {
+            csvPath = "/opt/local/hpds/columnMeta.csv";
+        }
+
+        this.processGroupedColumnMetas();
+        this.startProcessing();
+        try (BufferedReader br = new BufferedReader(new FileReader(csvPath))) {
+            final String[] previousConceptPath = {""};
```

Comment:
If we're only using `previousConceptPath[0]` do we need this to be a list? It can just be a string variable?